app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -89,9 +89,19 @@ def link_identity_to_service_provider
         current_user: current_user,
         ial: resolved_authn_context_int_ial,
         rails_session_id: session.id,
+        email_address_id: email_address_id,
       )
     end
 
+    def email_address_id
+      return session[:sp_email] unless session[:sp_email].nil?
+      sp = @authorize_form.service_provider.issuer
+      identity = current_user.identities.where(service_provider: sp)
+      email_id = identity.pluck('email_address_id')[0]
```

Comment:
I think `.first` reads better than `[0]`
```suggestion
      email_id = identity.pluck('email_address_id').first
```